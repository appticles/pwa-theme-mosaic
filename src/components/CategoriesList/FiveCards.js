import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridRow, GridColumn } from 'semantic-ui-react';

import CategoryCard from '../CategoryCard/index';

const FiveCards = props => (
  <Grid columns={2} divided>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[0]} key={Math.random()} />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[1]} tall={1} key={Math.random()} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[2]} tall={1} key={Math.random()} />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[3]} tall={1} key={Math.random()} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[4]} tall={1} key={Math.random()} />
      </GridColumn>
    </GridRow>
  </Grid>
);

FiveCards.propTypes = {
  categoriesList: PropTypes.array.isRequired,
};

export default FiveCards;
