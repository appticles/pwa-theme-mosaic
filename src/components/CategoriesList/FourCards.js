import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridRow, GridColumn } from 'semantic-ui-react';

import { categoryPropType } from '../../views/CategoriesCarousel/reducer';
import CategoryCard from '../CategoryCard/index';

const FourCards = props => (
  <Grid columns={2} divided>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[0]} tall={1} key={Math.random()} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[1]} tall={1} key={Math.random()} />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[2]} tall={1} key={Math.random()} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[3]} tall={1} key={Math.random()} />
      </GridColumn>
    </GridRow>
  </Grid>
);

FourCards.propTypes = {
  categoriesList: PropTypes.arrayOf(categoryPropType).isRequired,
};

export default FourCards;
