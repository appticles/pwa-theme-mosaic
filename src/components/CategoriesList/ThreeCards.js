import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridRow, GridColumn } from 'semantic-ui-react';

import CategoryCard from '../CategoryCard/index';

const ThreeCards = props => (
  <Grid columns={2} divided>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[0]} key={Math.random()} />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[1]} tall={1} key={Math.random()} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[2]} tall={1} key={Math.random()} />
      </GridColumn>
    </GridRow>
  </Grid>
);

ThreeCards.propTypes = {
  categoriesList: PropTypes.array.isRequired,
};

export default ThreeCards;
