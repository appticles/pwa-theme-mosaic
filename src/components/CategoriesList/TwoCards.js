import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridRow, GridColumn } from 'semantic-ui-react';

import CategoryCard from '../CategoryCard/index';

const TwoCards = props => (
  <Grid columns={2} divided>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[0]} tall={1} key={Math.random()} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[1]} tall={1} key={Math.random()} />
      </GridColumn>
    </GridRow>
  </Grid>
);

TwoCards.propTypes = {
  categoriesList: PropTypes.array.isRequired,
};

export default TwoCards;
