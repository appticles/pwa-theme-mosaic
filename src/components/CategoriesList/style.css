/* override semantic ui styles */
/* GRID */
.ui.grid {
  margin: 0;
}
.ui.grid > .row {
  justify-content: center;
  padding-top: 0;
  padding-bottom: 0;
}
.ui.grid > .row > .column {
  padding-left: 0;
  padding-right: 0;
}
.ui.divided.grid:not([class*="vertically divided"]) > .row > .column {
  box-shadow: none;
}
.slick-next,
.slick-prev {
  display: none !important;
}
.divided.two > .row > .column:only-child {
  width: 100vw;
}
