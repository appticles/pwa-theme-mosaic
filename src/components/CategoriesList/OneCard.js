import React from 'react';
import { Grid, GridRow } from 'semantic-ui-react';

import { categoryPropType } from '../../views/CategoriesCarousel/reducer';
import CategoryCard from '../CategoryCard/index';
import './style.css';

const OneCard = props => (
  <Grid className="full-row">
    <GridRow>
      <CategoryCard category={props.categoriesList[0]} key={Math.random()} />
    </GridRow>
  </Grid>
);

OneCard.propTypes = {
  categoriesList: categoryPropType.isRequired,
};
export default OneCard;
