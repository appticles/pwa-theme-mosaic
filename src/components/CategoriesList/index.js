import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './style.css';
import { categoryPropType } from '../../views/CategoriesCarousel/reducer';
import FiveCards from './FiveCards';
import FourCards from './FourCards';
import ThreeCards from './ThreeCards';
import TwoCards from './TwoCards';
import OneCard from './OneCard';

export default class CategoriesList extends Component {
  getCards() {
    switch (this.props.categoriesChunk.length % 5) {
    case 0:
      return <FiveCards categoriesList={this.props.categoriesChunk} />;
    case 1:
      return <OneCard categoriesList={this.props.categoriesChunk} />;
    case 2:
      return <TwoCards categoriesList={this.props.categoriesChunk} />;
    case 3:
      return <ThreeCards categoriesList={this.props.categoriesChunk} />;
    case 4:
      return <FourCards categoriesList={this.props.categoriesChunk} />;
    default:
      return null;
    }
  }

  render() {
    return <div>{this.props.categoriesChunk.length !== 0 ? this.getCards() : ''}</div>;
  }
}

CategoriesList.propTypes = {
  categoriesChunk: PropTypes.arrayOf(categoryPropType).isRequired,
};
