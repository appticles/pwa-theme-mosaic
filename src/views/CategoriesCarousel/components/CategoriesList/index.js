import React from 'react';
import PropTypes from 'prop-types';

import { categoryPropType } from '../../reducer';
import FiveCards from './FiveCards';
import FourCards from './FourCards';
import ThreeCards from './ThreeCards';
import TwoCards from './TwoCards';
import OneCard from './OneCard';

import './style.css';

const CategoriesList = (props) => {
  function getCards() {
    switch (props.categoriesChunk.length % 5) {
      case 0:
        return <FiveCards categoriesList={props.categoriesChunk} />;
      case 1:
        return <OneCard categoriesList={props.categoriesChunk} />;
      case 2:
        return <TwoCards categoriesList={props.categoriesChunk} />;
      case 3:
        return <ThreeCards categoriesList={props.categoriesChunk} />;
      case 4:
        return <FourCards categoriesList={props.categoriesChunk} />;
      default:
        return null;
    }
  }

  return <div>{props.categoriesChunk.length !== 0 ? getCards() : ''}</div>;
};

CategoriesList.propTypes = {
  categoriesChunk: PropTypes.arrayOf(categoryPropType).isRequired,
};

export default CategoriesList;
