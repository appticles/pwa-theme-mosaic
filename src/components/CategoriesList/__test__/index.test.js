import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CategoriesList from '../index';

describe('Categories list', () => {
  it('should render correctly 5 categories', () => {
    const mockCategoriesList = [
      {
        slug: 'mock-slug',
        id: 1,
        name: 'Some mock name',
      },
      {
        slug: 'mock-slug',
        id: 2,
        name: 'Some other mock name',
      },
      {
        slug: 'mock-slug',
        id: 3,
        name: 'Some name',
      },
      {
        slug: 'mock-slug',
        id: 4,
        name: 'Mock name',
      },
      {
        slug: 'mock-slug',
        id: 5,
        name: 'Some other other mock name',
      },
    ];

    const output = shallow(<CategoriesList categoriesChunk={mockCategoriesList} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should render correctly 4 categories', () => {
    const mockCategoriesList = [
      {
        slug: 'mock-slug',
        id: 1,
        name: 'Some mock name',
      },
      {
        slug: 'mock-slug',
        id: 2,
        name: 'Some other mock name',
      },
      {
        slug: 'mock-slug',
        id: 3,
        name: 'Some name',
      },
      {
        slug: 'mock-slug',
        id: 4,
        name: 'Mock name',
      },
    ];

    const output = shallow(<CategoriesList categoriesChunk={mockCategoriesList} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should render correctly 3 categories', () => {
    const mockCategoriesList = [
      {
        slug: 'mock-slug',
        id: 1,
        name: 'Some mock name',
      },
      {
        slug: 'mock-slug',
        id: 2,
        name: 'Some other mock name',
      },
      {
        slug: 'mock-slug',
        id: 3,
        name: 'Some name',
      },
    ];

    const output = shallow(<CategoriesList categoriesChunk={mockCategoriesList} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should render correctly 2 categories', () => {
    const mockCategoriesList = [
      {
        slug: 'mock-slug',
        id: 1,
        name: 'Some mock name',
      },
      {
        slug: 'mock-slug',
        id: 2,
        name: 'Some other mock name',
      },
    ];

    const output = shallow(<CategoriesList categoriesChunk={mockCategoriesList} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should render correctly one category', () => {
    const mockCategoriesList = [
      {
        slug: 'mock-slug',
        id: 1,
        name: 'Some mock name',
      },
    ];

    const output = shallow(<CategoriesList categoriesChunk={mockCategoriesList} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should render correctly no category', () => {
    const mockCategoriesList = [];

    const output = shallow(<CategoriesList categoriesChunk={mockCategoriesList} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
