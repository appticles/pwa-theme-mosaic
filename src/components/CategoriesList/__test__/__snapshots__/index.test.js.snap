// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories list should render correctly 1`] = `
<div>
  <ThreeCards
    categoriesList={
      Array [
        Object {
          "id": 1,
          "name": "Some mock name",
          "slug": "mock-slug",
        },
        Object {
          "id": 2,
          "name": "Some other mock name",
          "slug": "mock-slug",
        },
        Object {
          "id": 3,
          "name": "Some name",
          "slug": "mock-slug",
        },
        Object {
          "id": 4,
          "name": "Mock name",
          "slug": "mock-slug",
        },
        Object {
          "id": 5,
          "name": "Some other other mock name",
          "slug": "mock-slug",
        },
        Object {
          "id": 6,
          "name": "Some other other mock name",
          "slug": "mock-slug",
        },
        Object {
          "id": 7,
          "name": "Some other other mock name",
          "slug": "mock-slug",
        },
        Object {
          "id": 8,
          "name": "Some other other mock name",
          "slug": "mock-slug",
        },
      ]
    }
  />
</div>
`;
